pypesto/hierarchical/spline_approximation/solver.py

```diff
@@ -0,0 +1,875 @@
+        # plot_Boehm = True
+
+        # if plot_Boehm:
+        #     print("last opt values\n", [x.value for x in inner_parameters])
+        #     print("current opt values\n", results['x'])
+        #     print("current simulation\n", current_group_simulation)
+        #     print("intervals_per_sim\n", intervals_per_sim)
+        #     s = results['x']
+        #     xi = np.zeros(len(s))
+        #     for i in range(len(s)):
+        #         for j in range(i, len(s)):
+        #             xi[j] += s[i]
+
+        #     mapped_simulations = get_spline_mapped_simulations(
+        #         s,
+        #         current_group_simulation,
+        #         n_spline_pars,
+        #         distance_between_bases,
+        #         spline_bases,
+        #         intervals_per_sim
+        #     )
+        #     import matplotlib.pyplot as plt
+        #     plt.plot(current_group_simulation, group_measurements, 'bs')
+        #     plt.plot(spline_bases, xi, 'g')
+        #     plt.plot(spline_bases, xi, 'g.')
+        #     plt.plot(current_group_simulation, mapped_simulations, 'r^')
+        #     plt.show()
+
+        return results
+
+    def _rescale_spline_bases(self, sim_all: np.ndarray, N: int, K: int):
+        """Rescale the spline bases.
+
+        Before the optimization of the spline parameters, we have to fix the
+        spline bases to some values. We choose to scale them to the current
+        simulation. In case of simulations that are very close to each other,
+        we choose to scale closely around the average value of the simulations,
+        to avoid numerical problems (as we often divide by delta_c).
+
+        Parameters
+        ----------
+        sim_all:
+            The current simulation.
+        N:
+            The number of spline parameters.
+        K:
+            The number of simulations.
+
+        Returns
+        -------
+        distance_between_bases:
+            The distance between the spline bases.
+        spline_bases:
+            The rescaled spline bases.
+        intervals_per_sim:
+            List of indices of intervals each simulation belongs to.
+        """
+        min_idx = np.argmin(sim_all)
+        max_idx = np.argmax(sim_all)
+
+        min_all = sim_all[min_idx]
+        max_all = sim_all[max_idx]
+
+        n = np.ones(K)
+
+        # In case the simulation are very close to each other
+        # or even collapse into a single point (e.g. steady-state)
+        if max_all - min_all < MIN_SIM_RANGE:
+            average_value = (max_all + min_all) / 2
+            if average_value < (MIN_SIM_RANGE / 2):
+                delta_c = MIN_SIM_RANGE / (N - 1)
+                c = np.linspace(0, MIN_SIM_RANGE, N)
+            else:
+                delta_c = MIN_SIM_RANGE / (N - 1)
+                c = np.linspace(
+                    average_value - (MIN_SIM_RANGE / 2),
+                    average_value + (MIN_SIM_RANGE / 2),
+                    N,
+                )
```

Comment:
`delta_c` can be moved out of the `if` condition, since it's the same for both cases